package.json

```diff
@@ -2,43 +2,71 @@
   "name": "esutils",
   "description": "utility box for ECMAScript language tools",
   "homepage": "https://github.com/estools/esutils",
-  "main": "lib/utils.js",
+  "bugs": "https://github.com/estools/esutils/issues",
+  "main": "dist/esutils.min.js",
+  "module": "dist/esutils.esm.min.js",
```

Comment:
this isn’t a stand thing; ESM uses .mjs and the “exports” field (adding that field is also a breaking change)

Reply:
Yes, it is not standard but it is widely used and supported by bundlers like Webpack and Rollup.

As far as adding that field being a breaking change (if you mean `module` instead of `exports`), I'm not sure that would be the case here. It *can* be a breaking change when the `module` is pointing to untranspiled source.

There are advantages to pointing to untranspiled source, such as potentially minimizing package size and avoiding baking in dependencies, but that is not the case here where I am pointing to the rolled up (and Babelified) `dist` file.

And for those who might argue for changing to point to untranspiled source, on the one hand, this has disadvantages, besides of being a breaking change, of putting the work of building (and understanding of how to do so) on the consumer). Still, some might desire such a build, as with https://github.com/estools/esquery/issues/104 just filed against esquery, and for such use cases, it may indeed be better to add the breaking change to untranspiled source since it would bring the above-mentioned benefits (and those wishing to use a separate prebuilt distribution file could point to that file directly, whether the path to it is changed to be modulated by `exports` or not).

I would be happy adding the "mjs" extension (though we might clarify that it is not ESM that necessarily uses it, but Node which uses it), or a `type` change and/or the addition of `exports` (which would itself be a breaking change of sorts).

Reply:
Adding `module` isn't a breaking change because nothing *should* be using nonstandard fields; adding `exports` is what I was referring to.

Reply:
As far as the statement that "nothing should be using nonstandard fields", this is a good goal when equivalent ones are available, but otherwise I don't see *that* as a universally agreed upon goal. :)

[`module`](https://github.com/rollup/rollup/wiki/pkg.module) is not necessarily for Node only, nor is need for it removed with standard ESM implementations (it is an entry point for bundling). One might even argue it *is* a standard relative to the tools that have adopted its usage. And as far as its use relative to Node, Node has even made explicit it not opposing non-Node uses within its `package.json`, e.g.: re: `exports` custom conditions:

> Other conditions such as "browser", "electron", "deno", "react-native", etc. are ignored by Node.js but may be used by other runtimes or tools. Further restrictions, definitions or guidance on condition names may be provided in the future.

> https://nodejs.org/api/esm.html#esm_conditional_exports

But again, I'm happy to add proper Node support for non-bundled usage as well (if there is agreement on using `mjs` over `type: module`).

(As an aside re: standards, we might consider that Node itself [drifted away](https://github.com/nodejs/node-v0.x-archive/issues/5132#issuecomment-15432598) from the then CommonJS standard it had been following.)